File: server/types/videos.go
package types

import "time"

type VideoType struct {
	ID           string         `json:"id"`
	URL          string         `json:"url"`
	Title        string         `json:"title"`
	Description  string         `json:"description"`
	ChannelID    string         `json:"channel_id"`
	ChannelTitle string         `json:"channel_title"`
	PlaylistID   string         `json:"playlist_id"`
	PublishedAt  time.Time      `json:"published_at"`
	Thumbnail    ThumbnailType  `json:"thumbnail"`
	Tags         []string       `json:"tags"`
	Meta         *VideoMetaType `json:"meta"`
}

type ThumbnailType struct {
	URL    string `json:"url"`
	Width  uint32 `json:"width"`
	Height uint32 `json:"height"`
}

type VideoMetaType struct {
	Description  *LanguageStringType  `json:"description"`
	Correction   *LanguageStringType  `json:"correction"`
	TextSources  []LanguageStringType `json:"text_sources"`
	VideoSources []LanguageStringType `json:"video_sources"`
	AudioSources []LanguageStringType `json:"audio_sources"`
}

type LanguageStringType struct {
	German  string `json:"de-DE"`
	English string `json:"en-US"`
}

File: admin-client/src/types/videos.ts
export type VideoType = {
	id: string;
	url: string;
	title: string;
	description: string;
	channel_id: string;
	channel_title: string;
	playlist_id: string;
	published_at: string;
	thumbnail: {
		url: string;
		width: number;
		height: number;
	};
	tags: string[];
	meta: null | MetaType;
};

export type MetaType = {
	description: LanguageStringType | null;
	correction: LanguageStringType | null;
	text_sources: LanguageStringType[];
	video_sources: LanguageStringType[];
	audio_sources: LanguageStringType[];
};

export type LanguageStringType = {
	'de-DE': string;
	'en-US': string;
};
